src/matlab2tikz.m

```diff
@@ -892,37 +892,111 @@ case guitypes()
+            options = opts_add(options, grid, []);
         end
-    else
+        if hasMinorGrid(i)
+            grid     = [xyz{i}, 'minorgrids'];
+            options = opts_add(options, grid, []);
+        end
+    end
+
+    % Check for global grid options
+    if any(hasGrid)
+        gridOpts = [];
+        % Get the line style and translate it to pgfplots
+        [gridLS, isDefault] = getAndCheckDefault(...
+            'axes', handle, 'GridLineStyle', ':');
+        if ~isDefault || m2t.cmdOpts.Results.strict
+            gridOpts = translateLineStyle(gridLS);
+        end
+
+        % Get the color of the grid and translate it to pgfplots usable
+        % values
+        [gridColor, noColor] = getAndCheckDefault(...
+            'axes', handle, 'GridColor', [0.15, 0.15, 0.15]);
+        if ~noColor
+            [m2t, gridColor] = getColor(m2t, handle, gridColor, 'patch');
+            gridOpts = [gridOpts, ', ', gridColor];
+        end
+
+        % Get the alpha of the grid and translate it to pgfplots
+        % NOTE: The color conversion does not take alpha values into
+        % account. Therefore, we have to add it explicitely when using
+        % colored grids
+        [gridAlpha, noAlpha] = getAndCheckDefault(...
+            'axes', handle, 'GridAlpha', 0.1);
+        if ~noAlpha || ~noColor
+            gridOpts = [gridOpts, ', opacity =', num2str(gridAlpha)];
+        end
+
+        if ~isempty(gridOpts)
+            options = opts_add(options, 'grid style', sprintf('{%s}', gridOpts));
+        end
+    end
+
+    if any(hasMinorGrid)
+        % Get the line style and translate it to pgfplots
+        % NOTE: The default styles for minor grids differ between MATLAB
+        % and pgfplots. Therefore, always translate the style.
+        minorGridLS   = getAndCheckDefault(...
+            'axes', handle, 'MinorGridLineStyle', ':');
+        minorGridOpts = translateLineStyle(minorGridLS);
+
+        %Get the color of the grid and translate it to pgfplots usable
+        % values
+        [minorGridColor, noColor] = getAndCheckDefault(...
+            'axes', handle, 'MinorGridColor', [0.1, 0.1, 0.1]);
+        if ~noColor
+            [m2t, minorGridColor] = getColor(m2t, handle, minorGridColor, 'patch');
+            minorGridOpts = [minorGridOpts, ', ', minorGridColor];
+        end
+
+        % Get the alpha of the grid and translate it to pgfplots
+        % NOTE: The color conversion does not take alpha values into
+        % account. Therefore, we have to add it explicitely when using
+        % colored grids
+        [minorGridAlpha, noAlpha] = getAndCheckDefault(...
+            'axes', handle, 'MinorGridAlpha', 0.1);
+        if ~noAlpha || ~noColor
+            minorGridOpts = [minorGridOpts, ', opacity =', num2str(minorGridAlpha)];
+        end
+
+        if ~isempty(minorGridOpts)
+            options = opts_add(options, 'minor grid style', sprintf('{%s}', minorGridOpts));
```

Comment:
As for the major grid: remember to print the OptionArray `minorGridOpts` to string before passing this into the main `options`.